### ISSUE TITLE
Treat custom component as "img" primitive

### ISSUE THREAD

**Issue by aleclarson** [jsx-eslint/eslint-plugin-jsx-a11y#732]
I'm using [`ui-box`](https://github.com/segmentio/ui-box), which allows passing style props as JSX attributes.

Specifically, I want to provide a list of component names that should be treated as React DOM primitives. For example, I want my `Image` component (which renders as `img` element) to be treated as if I'm using `img` directly.

```jsx
import { Image } from './Image'

// No warning from eslint about missing "alt" 😢
<Image src={...} />
```

**Comment by ljharb** [comment]
See the docs for the rules:

 - https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/img-redundant-alt.md has a "components" option you can configure
 - https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/alt-text.md has a "img" option you can configure.

**Comment by aleclarson** [comment]
Oh nice! Sorry for not RTFM.
